Analysis/src/Coda3EventDecoder.cc

```diff
@@ -357,10 +358,17 @@ uint32_t Coda3EventDecoder::TBOBJ::Fill( const uint32_t* evbuffer,
 		uint32_t slen = *p & 0xffff;
 		if( slen != 2*(1 + (withRunInfo() ? 1 : 0) + (withTimeStamp() ? blkSize : 0)))
 			throw coda_format_error("Invalid length for Trigger Bank seg 1");
-		const auto* q = (const uint64_t*) (p + 1);
-		evtNum  = *q++;
-		runInfo = withRunInfo()   ? *q++ : 0;
-		evTS    = withTimeStamp() ? q    : nullptr;
+		memcpy(&evtNum, p + 1, sizeof(evtNum)); // uint64_t
+		if (withRunInfo()) {
+			memcpy(&runInfo, p + 3, sizeof(runInfo)); // uint64_t
+		} else {
+			runInfo = 0;
+		}
+		if (withTimeStamp()) {
+			evTS = reinterpret_cast<const uint64_t*>(p + 5); // uint64_t[blkSize]
```

Comment:
@Mrc1104 Good catch. Maybe this is a way to go.
```suggestion
		memcpy(&evtNum, p++, sizeof(evtNum)); // uint64_t
		if (withRunInfo()) {
			memcpy(&runInfo, p += 2, sizeof(runInfo)); // uint64_t
		} else {
			runInfo = 0;
		}
		if (withTimeStamp()) {
			evTS = reinterpret_cast<const uint64_t*>(p += 2); // uint64_t[blkSize]
```
I still think this is better handled in a way that then propagates back (or from) to where this code originated instead of diverging too much in little details. Or in a way that ensures in-memory alignment.

Reply:
> Looking at the [CODA Online Data Format](https://coda.jlab.org/drupal/system/files/eventbuilding.pdf) (see page 9, 11), the runInfo word should also come after the timestamp word(s). I haven't seen an evio file with the optional runInfo (I've checked evio files from NPS, LAD, and SBS and all trigger tags that I have seen have been 0xff21), and a cursory glance through the TI library does not show an obvious way to 'enable' the runInfo. So I am not sure how to verify this. I'll email [JLab's DAQ group](https://coda.jlab.org/drupal/content/about-us) and see if I can get some guidance.

Per Dave Abbott, we can enable runInfo for the event builder using jcedit. Doing so, I can confirm that the runInfo is indeed after the timestamp word(s). 

```suggestion
               const uint32_t *q = (p+1);
		memcpy(&evtNum, q++, sizeof(evtNum)); // uint64_t
		if (withTimeStamp()) {
		       // ++q is more clear to me than q+=1
			evTS = reinterpret_cast<const uint64_t*>(++q); // uint64_t[blkSize]
			q+= blksize-1;
		} else {
		        evTS = nullptr;
		}
	        if (withRunInfo()) {
			memcpy(&runInfo, p+=2, sizeof(runInfo)); // uint64_t
		} else {
			runInfo = 0;
		}
		p += slen + 1;
```

### runInfo disabled:
```
p[0] = 10a0004
p[1] = f72
p[2] = 0
p[3] = 12345678
p[4] = 0
p[5] = 1850001
p[6] = 1
p[7] = 1010002
p[8] = 12345678
Segment1
p[0] = 10a0004
p[1] = f72
p[2] = 0
p[3] = 12345678
Has runInfo? false              Has timeStamp? true
tag: 0xff21             runInfo: 0x0    (high 32bit: 0) (low 32bit: 0)          evTS: 0x12345678                evNum: f72
```

### runInfo enabled:
```
p[0] = 10a0006
p[1] = f72
p[2] = 0
p[3] = 12345678
p[4] = 0
p[5] = 1
p[6] = 3
p[7] = 1850001
p[8] = 1
p[9] = 1010002
p[a] = 12345678
Segment1
p[0] = 10a0006
p[1] = f72
p[2] = 0
p[3] = 12345678
p[4] = 0
p[5] = 1
p[6] = 3
Has runInfo? true               Has timeStamp? true
tag: 0xff23             runInfo: 0x300000001    (high 32bit: 3) (low 32bit: 1)          evTS: 0x12345678                evNum: f72
```